src/Stryker.Core/Stryker.Core/Initialisation/Buildalyzer/IAnalyzerResultExtensions.cs

```diff
@@ -97,16 +96,20 @@ private sealed class AnalyzerAssemblyLoader : IAnalyzerAssemblyLoader
             public Assembly LoadFromPath(string fullPath) => Assembly.LoadFrom(fullPath);
         }
 
+        internal static NuGetFramework GetNuGetFramework(this IAnalyzerResult analyzerResult)
+        {
+            return NuGetFramework.Parse(analyzerResult.TargetFramework ?? "");
+        }
+
+        internal static bool TargetsFullFramework(this IAnalyzerResult analyzerResult)
+        {
+            return GetNuGetFramework(analyzerResult).IsDesktop();
+        }
+
+        [Obsolete("Please consider using the NuGetFramework type instead (from the 'NuGet.Frameworks' package)", error: true)]
```

Comment:
Marking things as Obsolete is not something we generally do. It might be a good idea if we currently cannot easily replace all usages, but then I would expect to keep the unit test for now.

Reply:
I'm not sure to understand what you mean. Should I completely delete the now unused methods or should I just keep them around for compatibility even if they are not referenced anymore?

I initially chose to keep them and mark them obsolete since the `IAnalyzerResultExtensions` class and the `Framework` enum are public types. If someone is using those types when consuming the `stryker` NuGet package , deleting them would leave them in the dark. With the `Obsolete` attribute, a clear error message hints at a solution for replacing the removed types and methods.

Anyway, I'll update the pull request to either keep the methods or delete them as you see fit.

Reply:
I did not check if all usages were already replaced. 

If they are, you can remove the methods and classes instead of marking as obsolete. As far as I know no one else currently uses the Stryker core package and I am fine with breaking in this way until we have actual consumers.

If they are not yet all replaced (within Stryker), marking as obsolete seems fine to me, but then we should keep the unit tests.